en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/observability/distributed-tracing.md

```diff
@@ -0,0 +1,112 @@
+    zipkin:
+      image: openzipkin/zipkin
+      container_name: zipkin
+      ports:
+        - "9411:9411"
+    ```
+
+### Jaeger
+
+Follow these steps to configure WSO2 Choreo Connect with Jaeger,
+
+1. Add below configuration to the `config.toml`  
+    ```toml
+    [enforcer.tracing]
+      enabled = true
+      type = "jaeger"
+      [enforcer.tracing.configProperties]
+          endpoint = "http://jaeger:14268/api/traces"
+          instrumentationName = "CHOREO-CONNECT"
+          maximumTracesPerSecond = "2"
+    ```  
+1. Start Choreo Connect deployment.
+1. [Create and Deploy an API]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/quick-start-guide/quick-start-guide-docker-with-apim/)
+1. Invoke the newly created API and open Jaeger UI to view the traces. Ex: <http://localhost:16686>
+
+You will be able to browse through the request traces and expand each trace to view complete trace details.
+
+![Jaeger Trace]({{base_path}}/assets/img/deploy/mgw/trace-jaeger.png)
+
+### Zipkin
+
+Follow these steps to configure WSO2 Choreo Connect with Zipkin,
+
+1. Add below configuration to the `config.toml`  
+    ```toml
+    [enforcer.tracing]
+      enabled = true
+      type = "azure"
+      [enforcer.tracing.configProperties]
+        endpoint = "http://zipkin:9411/api/v2/spans"
+        instrumentationName = "CHOREO-CONNECT"
+        maximumTracesPerSecond = "2"
+    ```  
+1. Start Choreo Connect deployment.
+1. [Create and Deploy an API]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/quick-start-guide/quick-start-guide-docker-with-apim/)
+1. Invoke the newly create API and open Zipkin UI to view the traces. Ex: <http://localhost:9411>
+1. Filter traces by `serviceName=choreo_connect` query.
+
+You will be able to see all traces. Detailed trace view will look like below.
+
+![Zipkin Trace]({{base_path}}/assets/img/deploy/mgw/trace-zipkin.png)
+
+### Azure Application Insights
+
+Follow these steps to configure WSO2 Choreo Connect with Azure Application Insights,
+
+1. First you need to obtain the `ConnectionString` from the azure portal.
+    * Log into azure portal
+    * Copy the `Connection String` from overview page of `Application Insights` resource. Ex: `InstrumentationKey=ab71943f-xxxx-xxxx-xxxx-fb2eb69ae11d;IngestionEndpoint=https://region.applicationinsights.azure.com/`
+1. Set above Connection String as an environment variable in enforcer container.
+
+    ```yaml tab="Example"
+    enforcer:
+      ...
+      environment:
+        - APPLICATIONINSIGHTS_CONNECTION_STRING=InstrumentationKey=ab71943f-xxxx-xxxx-xxxx-fb2eb69ae11d;IngestionEndpoint=https://xxxxxx.applicationinsights.azure.com/
+    ```
+
+1. Add below configuration to the `config.toml`  
+    ```toml
+    [enforcer.tracing]
+      enabled = true
+      type = "azure"
+      [enforcer.tracing.configProperties]
+        connectionString = "$env{APPLICATIONINSIGHTS_CONNECTION_STRING}"
```

Comment:
WIthout complicating things shall we define the connection string inline in the guide. Users are able to provide any configs as envs and we dont need to specify here.

Reply:
Fixed with 629aec8